[receiver/kubeletstats] Add feature gate for cpu utilization metrics' deprecation (#35139)
open-telemetry/opentelemetry-collector-contrib: receiver/kubeletstatsreceiver/README.md

    resources: ["nodes/proxy"]
    verbs: ["get"]
```

### Warning about metrics' deprecation

> The following metrics will be removed in a future version:
> - `k8s.node.cpu.utilization` (replaced by `k8s.node.cpu.usage`)
> - `k8s.pod.cpu.utilization` (replaced by `k8s.pod.cpu.usage`)
> - `container.cpu.utilization` (replaced by `container.cpu.usage`)

Reviewer: ```suggestion
The following metrics will be renamed in a future version:
- `k8s.node.cpu.utilization` (renamed to `k8s.node.cpu.usage`)
- `k8s.pod.cpu.utilization` (renamed to `k8s.pod.cpu.usage`)
- `container.cpu.utilization` (renamed to `container.cpu.usage`)
```

Using `rename` instead of `replace` will hopefully up users at ease as they can trust the new metric is exactly the same.

Users can use the following metrics instead:
- `k8s.node.cpu.usage`
- `k8s.pod.cpu.usage`

> - `container.cpu.usage`

Reviewer: Since you already listed the `replaced by` in the previous section, I think this should be removed.
